### PR TITLE
3.0 preview dev

### PR DESCRIPTION
**Pull request by e111077**
I had to create a big google-youtube; might as well've migrated it

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,217 +1,98 @@
...
-## &lt;app-location&gt;
-
-`app-location` is an element that provides synchronization between the
-browser location bar and the state of an app. When created, `app-location`
-elements will automatically watch the global location for changes. As changes
-occur, `app-location` produces and updates an object called `route`. This
-`route` object is suitable for passing into a `app-route`, and other similar
-elements.
+See: [Documentation](https://www.webcomponents.org/element/@polymer/app-route),
+  [Large Demo](https://www.webcomponents.org/element/@polymer/app-route/demo/demo/index.html),
```
Comment (notwaldorf) [inline README.md]: Nice!

File: app-route.js [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+import '@polymer/polymer/polymer-legacy.js';
+
+import {Polymer} from '@polymer/polymer/lib/legacy/polymer-fn.js';
+
 /**
 `app-route` is an element that enables declarative, self-describing routing
 for a web app.
 
-> *n.b. app-route is still in beta. We expect it will need some changes. We're counting on your feedback!*
+> *n.b. app-route is still in beta. We expect it will need some changes. We're
```
Comment (notwaldorf) [inline app-route.js]: Is this still true? Will this stop being true? Maybe we should just delete this

File: demo/data-loading-demo.html [hunk 3]
```diff
@@ -15,44 +15,20 @@
...
-
-document.head.appendChild($_documentContainer.content);
-</script>
+  <script src="https://unpkg.com/@webcomponents/webcomponentsjs@^2.0.0/webcomponents-loader.js"></script>
+  <script type="module">
+    import '@polymer/iron-demo-helpers/url-bar.js';
+    import './data-loading-demo/flickr-search-demo.js';
+    import {html} from '@polymer/polymer/lib/utils/html-tag.js';
+    const template =
+        html`<style is="custom-style" include="demo-pages-shared-styles">
```
Comment (notwaldorf) [inline demo/data-loading-demo.html]: doesn't this need to be wrapped in a `<custom-style>` for it to actually work in 2.0+?

File: demo/index.html [hunk 4]
```diff
@@ -14,21 +14,9 @@
 -->
 <html>
 <head>
-<script src="../../../@webcomponents/webcomponentsjs/webcomponents-bundle.js"></script>
-
+  <script src="https://unpkg.com/@webcomponents/webcomponentsjs@^2.0.0/webcomponents-loader.js"></script>
+  <script src="https://www.youtube.com/iframe_api"></script>
```
Comment (notwaldorf) [inline demo/index.html]: indentation looks weird compared to the `<title>` below

File: demo/youtube-demo/google-youtube.js [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+ */
+class GoogleYoutube extends PolymerElement {
+  static get template() {
+    return html`
+      <style>
+        :host {
+          display: block;
+        }
+
+        :host([fluid]) {
```
Comment (notwaldorf) [inline demo/youtube-demo/google-youtube.js]: Are we using this attribute anywhere? I didn't find it in the demo with a find-in-page but maybe I'm holding it wrong. If we don't need it for the demo, maybe just nuke it to keep it simplerer?

File: demo/youtube-demo/google-youtube.js [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        type: String,
+        observer: '_videoIdChanged',
+      },
+
+      state: {
+        type: Number,
+        notify: true,
+      },
+
+      currenttime: {
```
Comment (notwaldorf) [inline demo/youtube-demo/google-youtube.js]: this is a bit 🤢. Can it be `currentTime`?

File: demo/youtube-demo/video-viewer.js [hunk 7]
```diff
@@ -60,24 +61,38 @@ Polymer({
     provides this route decides where this element lives in the URL space. In
     this case, the parent, which uses hashes, matches #/video and hence this
     element lives in <App serving point>?querParams#/video/:vid -->
```
Comment (notwaldorf) [inline demo/youtube-demo/video-viewer.js]: tiny type: i think this is `?queryParams`

File: demo/youtube-demo/youtube-lite.js [hunk 8]
```diff
@@ -20,10 +21,11 @@ Note that this element is totally agnostic to routing!
   then delete this comment!
 */
 import '@polymer/polymer/polymer-legacy.js';
+import './google-youtube.js';
```
Comment (notwaldorf) [inline demo/youtube-demo/youtube-lite.js]: There's still a `FIXME` comment above this but I can't comment on that line 🙄

File: test/app-location.html [hunk 9]
```diff
@@ -12,13 +12,8 @@
...
   <title>app-location</title>
 
-  <script src="../../../@webcomponents/webcomponentsjs/webcomponents-bundle.js"></script>
-  <script src="../../../wct-browser-legacy/browser.js"></script>
-  <script src="../../../url/url.js"></script>
-
-  <script type="module" src="../../polymer/polymer-legacy.js"></script>
-  <script type="module" src="../../iron-test-helpers/mock-interactions.js"></script>
-  <script type="module" src="../app-location.js"></script>
+  <script src="../node_modules/@webcomponents/webcomponentsjs/webcomponents-bundle.js"></script>
```
Comment (notwaldorf) [inline test/app-location.html]: Im assuming most of these test diffs are whitespace right?

File: app-route.js (hunk 2 above)
Comment (e111077) [reply app-route.js]: I totes just ignored this section oops

File: demo/data-loading-demo.html (hunk 3 above)
Comment (e111077) [reply demo/data-loading-demo.html]: this style is never acutally used

File: demo/youtube-demo/google-youtube.js (hunk 6 above)
Comment (e111077) [reply demo/youtube-demo/google-youtube.js]: this was the default in google-youtube. I think I may have caught all the places I've used it

File: test/app-location.html (hunk 9 above)
Comment (e111077) [reply test/app-location.html]: yup. See this link:

**Comment by e111077** [comment]
Ready for another review! 😃

**Comment by notwaldorf** [comment]
> Project is not formatted. Please run "npm run format". 😂

**Comment by notwaldorf** [comment]
(you did this to us)

File: README.md [hunk 10]
```diff
@@ -1,217 +1,98 @@
...
+  <head>
+    <script type="module">
+      import '@polymer/app-route/app-location.js';
+      import '@polymer/app-route/app-route.js';
+    </script>
+  </head>
+  <body>
+    <app-location route="{{route}}"></app-location>
+    <app-route
+        route="{{route}}"
```
Comment (notwaldorf) [inline README.md]: This doesn't work in an html file like this...

Comment (e111077) [reply README.md]: ah

Comment (e111077) [reply README.md]: fixed

**Pull request merged**